aeron-test-support/src/main/java/io/aeron/test/DataCollector.java

```diff
@@ -146,14 +169,67 @@ public Path dumpData(final String destinationDir)
             .collect(toList());
     }
 
-    public List<Path> errorLogFiles()
+    /**
+     * Find all the clustered service mark files.
+     *
+     * @return list of paths to the clustered service mark files.
+     */
+    public List<Path> clusterServiceMarkFiles()
     {
         return this.locations.stream()
-            .map(p -> p.resolve(TestNode.CLUSTER_ERROR_FILE))
-            .filter(Files::exists)
+            .flatMap(DataCollector::list)
+            .filter(ClusterMarkFile::isServiceMarkFile)
+            .collect(toList());
+    }
+
+    /**
+     * Find all the consensus module mark files.
+     *
+     * @return list of paths to the consensus module mark files
+     */
+    public List<Path> consensusModuleMarkFiles()
+    {
+        return this.locations.stream()
+            .flatMap(DataCollector::list)
+            .filter(ClusterMarkFile::isConsensusModuleMarkFile)
             .collect(toList());
     }
 
+    /**
+     * Find all the archive mark files.
+     *
+     * @return list of paths to the archive mark files
+     */
+    public List<Path> archiveMarkFiles()
+    {
+        return this.locations.stream()
+            .flatMap(DataCollector::list)
+            .filter(ArchiveMarkFile::isArchiveMarkFile)
+            .collect(toList());
+    }
+
+    /**
+     * Gets a collection of all registered locations.
+     *
+     * @return list of all locations.
+     */
+    public Collection<Path> allLocations()
+    {
+        return locations;
+    }
+
+    private static Stream<Path> list(final Path p)
+    {
+        try
+        {
+            return Files.list(p);
```

Comment:
`DataCollector` contains not just directories but also the individual files and thus `Files.list(p)` will fail when one is encountered. This can be seen from the failing C tests which add stdout/stderr files to the `DataCollector`.

You could use `java.nio.file.Files#find` method instead which works fine if `start` is a file.
To list all of the files with a particular name you can use `java.nio.file.Files#find` method starting at the

Reply:
Done.